sample_xblocks/basic/problem.py

```diff
@@ -117,13 +117,29 @@ def student_view(self, context=None):
                     }
                 }
 
-                function handleCheckResults(results) {
-                    $.each(results.submitResults || {}, function(input, result) {
-                        callIfExists(runtime.childMap(element, input), 'handleSubmit', result);
-                    });
-                    $.each(results.checkResults || {}, function(checker, result) {
-                        callIfExists(runtime.childMap(element, checker), 'handleCheck', result);
-                    });
+                function handleCheckResults(data) {
+                    return function(results) {
+                        $.each(results.submitResults || {}, function(input, result) {
+                            callIfExists(runtime.childMap(element, input), 'handleSubmit', result);
+                        });
+                        answers = {};
```

Comment:
These variables aren't being declared, causing them to leak into the global namespace.

```javascript
var answers = {};
var correct_map = {};
var key;
var event;
```